ColorCoding.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ColorCoding.h"
+namespace TelCoColorCoder
+{
+
+ const char* MajorColorNames[] = {
+        "White", "Red", "Black", "Yellow", "Violet"
+    };
+    int numberOfMajorColors =
+        sizeof(MajorColorNames) / sizeof(MajorColorNames[0]);
+    const char* MinorColorNames[] = {
+        "Blue", "Orange", "Green", "Brown", "Slate"
+    };
+    int numberOfMinorColors = sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
+ 
+    ColorPair::ColorPair(MajorColor major, MinorColor minor) :
+        majorColor(major), minorColor(minor)
+    {}
+    MajorColor ColorPair::getMajor() {
+        return majorColor;
+    }
+    MinorColor ColorPair::getMinor() {
+        return minorColor;
+    }
+    std::string ColorPair::ToString() {
+        std::string colorPairStr = MajorColorNames[majorColor];
+        colorPairStr += " ";
+        colorPairStr += MinorColorNames[minorColor];
+        return colorPairStr;
+    }
+ 
+    ColorPair GetColorFromPairNumber(int pairNumber) {
+        int zeroBasedPairNumber = pairNumber - 1;
+        MajorColor majorColor = 
+            (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+        MinorColor minorColor =
+            (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+        return ColorPair(majorColor, minorColor);
+    }
+    int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
+        return major * numberOfMinorColors + minor + 1;
+    }
+
+    std::string GetColorCoding(){
+        std::string ColorCodingPrint = "Color Coding :\n";
+        for(int i = 1; i <= 25; i++)
```

Comment:
Don't hard code numbers like `25`. it's hard to keep track and update them when changes happen